boot/bootutil/include/bootutil/crypto/hmac_sha256.h

```diff
@@ -22,7 +22,9 @@
     #include <stddef.h>
     #include <mbedtls/cmac.h>
     #include <mbedtls/md.h>
-    #include <mbedtls/md_internal.h>
+    #if MBEDTLS_VERSION_NUMBER < 0x03000000
+        #include <mbedtls/md_internal.h>
+    #endif
```

Comment:
I think it is okay to completely remove this include from here.
As I see it there has been no reason for it to be included for a while (in hmac_sha256.h).

```suggestion
```